### DISCUSSION TITLE
run_Noahmp

### DISCUSSION THREAD

**Discussion by anlei-code** [NCAR/noahmp#167] [Q&A]
I run the noahmp with the geo_em.d01.nc which is made in WRF/WPS by myself,but it's results is not good,some vars disappeared,like LH ,FIRA,EMISS.It seems that the energy part is not run,But,i use the geo_em.d01.nc which is given in the tutorial, it's results seem right.SO, i think the error is the geo_em.d01.nc-making,i have added the optional data to the WPS_GEOG to consistent with the reference geo_em.d01.nc,but it not useful.Could you share a paper about how to make geo_em.d01.nc in wps for noahmp to me,i don't know what error is.
![图片](https://github.com/user-attachments/assets/4381674e-cc5a-4d1c-8eab-1ba409173ca9)
![图片](https://github.com/user-attachments/assets/a248abcd-7f16-4ced-af6b-25ebfcc3a651)
![图片](https://github.com/user-attachments/assets/050aeffd-f134-4137-a996-673dcd2e4215)
![图片](https://github.com/user-attachments/assets/79d82401-9585-44b6-91ae-26aeb25e7d40)
![图片](https://github.com/user-attachments/assets/dcd0efb9-8446-4ca9-85f8-edbc48520526)

**Comment by cenlinhe** [comment]
You should be able to use the geo_em file generated by the default WPS. There is no special requirement for generating Noah-MP geo_em file because it works with default WRF/WPS geo_em file. The only thing that needs attention is that if you want to activate Noah-MP crop, irrigation, or MMF groundwater schemes in your simulation, you will need to use the geogrid/GEOGRID.TBL.ARW_noahmp instead of the default GEOGRID.TBL.ARW look up table before running geogrid.exe. If you do not activate those schemes in your namelist.hrldas, then this is not relevant.
Also, when you use your own geo_em file, did you also re-generate all the input forcing files based on your own geo_em file? you need to have the forcing files consistent with your geo_em file domain setup.

**Comment by anlei-code** [comment]
I have tried the default WPS to make geo_em.nc to run the noahmp,but the results not changed,also not right.But i have find that the reference geo_em.d01.nc's landsue is not consistent with mine,mybe problem is this?
![图片](https://github.com/user-attachments/assets/dc96c469-b9f0-40de-aa41-2d1af8617c30)

**Reply by cenlinhe** [reply]
I think your land type map is fine. The only difference is that the reference geo_em file uses 20-category MODIS type (with both ocean and lakes identified as 17), while your own one uses the 21-category MODIS type with the lakes identified as 21. This won't affect the Noah-MP simulations. The model will skip all the water points.

**Reply by anlei-code** [reply]
thankyou for your help,maybe i need to change reinstall the WRF/WPS.

**Comment by anlei-code** [comment]
[relate_file.zip](https://github.com/user-attachments/files/18150358/relate_file.zip)

**Comment by tslin2** [comment]
Which version of WPS and landcover data did you use?

**Reply by anlei-code** [reply]
version of WRF is 4.0,the WPS is 4.3.1。the landcover is default (modies_landcover_20classes_with_lake).maybe i need to reinstall the WRF/WPS with other versions.
